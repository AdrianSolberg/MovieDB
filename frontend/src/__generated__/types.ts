export type Maybe<T> = T | null;
export type InputMaybe<T> = Maybe<T>;
export type Exact<T extends { [key: string]: unknown }> = { [K in keyof T]: T[K] };
export type MakeOptional<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]?: Maybe<T[SubKey]> };
export type MakeMaybe<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]: Maybe<T[SubKey]> };
export type MakeEmpty<T extends { [key: string]: unknown }, K extends keyof T> = { [_ in K]?: never };
export type Incremental<T> = T | { [P in keyof T]?: P extends ' $fragmentName' | '__typename' ? T[P] : never };
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: { input: string; output: string; }
  String: { input: string; output: string; }
  Boolean: { input: boolean; output: boolean; }
  Int: { input: number; output: number; }
  Float: { input: number; output: number; }
  Date: { input: any; output: any; }
};

export type Movie = {
  _id: Scalars['Int']['output'];
  title: Scalars['String']['output'];
  vote_average: Scalars['Float']['output'];
  vote_count: Scalars['Int']['output'];
  status: Scalars['String']['output'];
  release_date: Scalars['Date']['output'];
  revenue: Scalars['Float']['output'];
  runtime: Scalars['Int']['output'];
  backdrop_path?: Maybe<Scalars['String']['output']>;
  budget: Scalars['Int']['output'];
  homepage?: Maybe<Scalars['String']['output']>;
  imdb_id?: Maybe<Scalars['String']['output']>;
  original_language: Scalars['String']['output'];
  original_title: Scalars['String']['output'];
  overview: Scalars['String']['output'];
  popularity: Scalars['Float']['output'];
  poster_path?: Maybe<Scalars['String']['output']>;
  tagline?: Maybe<Scalars['String']['output']>;
  genres: Array<Scalars['String']['output']>;
  production_companies: Array<Scalars['String']['output']>;
  production_countries: Array<Scalars['String']['output']>;
  spoken_languages: Array<Scalars['String']['output']>;
  keywords: Array<Scalars['String']['output']>;
};

export type Filters = {
  Genre: Array<Scalars['String']['output']>;
  Rating: Array<Scalars['String']['output']>;
  Decade: Array<Scalars['String']['output']>;
  Status: Array<Scalars['String']['output']>;
  Runtime: Array<Scalars['String']['output']>;
};

export type FiltersInput = {
  Genre: Array<Scalars['String']['input']>;
  Rating: Array<Scalars['String']['input']>;
  Decade: Array<Scalars['String']['input']>;
  Status: Array<Scalars['String']['input']>;
  Runtime: Array<Scalars['String']['input']>;
};

export enum SortingType {
  NewestFirst = 'NEWEST_FIRST',
  OldestFirst = 'OLDEST_FIRST',
  BestRated = 'BEST_RATED',
  WorstRated = 'WORST_RATED',
  LongestRuntime = 'LONGEST_RUNTIME',
  ShortestRuntime = 'SHORTEST_RUNTIME'
}

export type Query = {
  movie?: Maybe<Movie>;
  movies: Array<Movie>;
  filters?: Maybe<Filters>;
};


export type QueryMovieArgs = {
  id: Scalars['Int']['input'];
};


export type QueryMoviesArgs = {
  skip?: InputMaybe<Scalars['Int']['input']>;
  limit?: InputMaybe<Scalars['Int']['input']>;
  filters?: InputMaybe<FiltersInput>;
  sortOption?: InputMaybe<SortingType>;
  search?: InputMaybe<Scalars['String']['input']>;
};

export type GetMovieQueryVariables = Exact<{
  movieId: Scalars['Int']['input'];
}>;


export type GetMovieQuery = { __typename?: 'Query', movie?: { __typename?: 'Movie', _id: number, title: string, vote_average: number, vote_count: number, status: string, release_date: any, revenue: number, runtime: number, backdrop_path?: string | null, budget: number, homepage?: string | null, imdb_id?: string | null, original_language: string, original_title: string, overview: string, popularity: number, poster_path?: string | null, tagline?: string | null, genres: Array<string>, production_companies: Array<string>, production_countries: Array<string>, spoken_languages: Array<string>, keywords: Array<string> } | null };

export type GetMoviesQueryVariables = Exact<{
  skip?: InputMaybe<Scalars['Int']['input']>;
  limit?: InputMaybe<Scalars['Int']['input']>;
  filters?: InputMaybe<FiltersInput>;
  sortOption?: InputMaybe<SortingType>;
  search?: InputMaybe<Scalars['String']['input']>;
}>;


export type GetMoviesQuery = { __typename?: 'Query', movies: Array<{ __typename?: 'Movie', _id: number, title: string, vote_average: number, release_date: any, runtime: number, poster_path?: string | null }> };

export type GetFiltersQueryVariables = Exact<{ [key: string]: never; }>;


export type GetFiltersQuery = { __typename?: 'Query', filters?: { __typename?: 'Filters', Genre: Array<string>, Rating: Array<string>, Decade: Array<string>, Status: Array<string>, Runtime: Array<string> } | null };
