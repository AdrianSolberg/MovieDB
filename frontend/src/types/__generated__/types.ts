export type Maybe<T> = T | null;
export type InputMaybe<T> = Maybe<T>;
export type Exact<T extends { [key: string]: unknown }> = {
    [K in keyof T]: T[K];
};
export type MakeOptional<T, K extends keyof T> = Omit<T, K> & {
    [SubKey in K]?: Maybe<T[SubKey]>;
};
export type MakeMaybe<T, K extends keyof T> = Omit<T, K> & {
    [SubKey in K]: Maybe<T[SubKey]>;
};
export type MakeEmpty<
    T extends { [key: string]: unknown },
    K extends keyof T,
> = { [_ in K]?: never };
export type Incremental<T> =
    | T
    | {
          [P in keyof T]?: P extends " $fragmentName" | "__typename"
              ? T[P]
              : never;
      };
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
    ID: { input: string; output: string };
    String: { input: string; output: string };
    Boolean: { input: boolean; output: boolean };
    Int: { input: number; output: number };
    Float: { input: number; output: number };
    DateTime: { input: Date; output: Date };
};

export type Movie = {
    __typename?: "Movie";
    _id: Scalars["Int"]["output"];
    title: Scalars["String"]["output"];
    vote_average: Scalars["Float"]["output"];
    vote_count: Scalars["Int"]["output"];
    status: Scalars["String"]["output"];
    release_date: Scalars["DateTime"]["output"];
    revenue: Scalars["Float"]["output"];
    runtime: Scalars["Int"]["output"];
    backdrop_path?: Maybe<Scalars["String"]["output"]>;
    budget: Scalars["Int"]["output"];
    homepage?: Maybe<Scalars["String"]["output"]>;
    imdb_id?: Maybe<Scalars["String"]["output"]>;
    original_language: Scalars["String"]["output"];
    original_title: Scalars["String"]["output"];
    overview: Scalars["String"]["output"];
    popularity: Scalars["Float"]["output"];
    poster_path?: Maybe<Scalars["String"]["output"]>;
    tagline?: Maybe<Scalars["String"]["output"]>;
    genres: Array<Scalars["String"]["output"]>;
    production_companies: Array<Scalars["String"]["output"]>;
    production_countries: Array<Scalars["String"]["output"]>;
    spoken_languages: Array<Scalars["String"]["output"]>;
    keywords: Array<Scalars["String"]["output"]>;
    reviews: Array<Review>;
};

export type Review = {
    __typename?: "Review";
    _id: Scalars["ID"]["output"];
    movie: Movie;
    username: Scalars["String"]["output"];
    rating: Scalars["Int"]["output"];
    comment: Scalars["String"]["output"];
    date: Scalars["DateTime"]["output"];
};

export type Filters = {
    __typename?: "Filters";
    Genre: Array<Scalars["String"]["output"]>;
    Rating: Array<Scalars["String"]["output"]>;
    Decade: Array<Scalars["String"]["output"]>;
    Status: Array<Scalars["String"]["output"]>;
    Runtime: Array<Scalars["String"]["output"]>;
};

export type FiltersInput = {
    Genre: Array<Scalars["String"]["input"]>;
    Rating: Array<Scalars["String"]["input"]>;
    Decade: Array<Scalars["String"]["input"]>;
    Status: Array<Scalars["String"]["input"]>;
    Runtime: Array<Scalars["String"]["input"]>;
};

export enum SortingType {
    MOST_POPULAR = "MOST_POPULAR",
    NEWEST_FIRST = "NEWEST_FIRST",
    OLDEST_FIRST = "OLDEST_FIRST",
    BEST_RATED = "BEST_RATED",
    WORST_RATED = "WORST_RATED",
    LONGEST_RUNTIME = "LONGEST_RUNTIME",
    SHORTEST_RUNTIME = "SHORTEST_RUNTIME",
}

export type Query = {
    __typename?: "Query";
    movie?: Maybe<Movie>;
    movies: Array<Movie>;
    filters: Filters;
    latestReviews: Array<Review>;
    userReviews: Array<Review>;
};

export type QuerymovieArgs = {
    id: Scalars["Int"]["input"];
};

export type QuerymoviesArgs = {
    skip?: InputMaybe<Scalars["Int"]["input"]>;
    limit?: InputMaybe<Scalars["Int"]["input"]>;
    filters?: InputMaybe<FiltersInput>;
    sortOption?: InputMaybe<SortingType>;
    search?: InputMaybe<Scalars["String"]["input"]>;
};

export type QuerylatestReviewsArgs = {
    skip?: InputMaybe<Scalars["Int"]["input"]>;
    limit?: InputMaybe<Scalars["Int"]["input"]>;
};

export type QueryuserReviewsArgs = {
    username: Scalars["String"]["input"];
    skip?: InputMaybe<Scalars["Int"]["input"]>;
    limit?: InputMaybe<Scalars["Int"]["input"]>;
};

export type Mutation = {
    __typename?: "Mutation";
    addReview: Movie;
    deleteReview: Movie;
};

export type MutationaddReviewArgs = {
    movie_id: Scalars["Int"]["input"];
    username: Scalars["String"]["input"];
    rating: Scalars["Int"]["input"];
    comment: Scalars["String"]["input"];
};

export type MutationdeleteReviewArgs = {
    _id: Scalars["ID"]["input"];
};

export type GetMoviesQueryVariables = Exact<{
    skip?: InputMaybe<Scalars["Int"]["input"]>;
    limit?: InputMaybe<Scalars["Int"]["input"]>;
    filters?: InputMaybe<FiltersInput>;
    sortOption?: InputMaybe<SortingType>;
    search?: InputMaybe<Scalars["String"]["input"]>;
}>;

export type GetMoviesQuery = {
    __typename?: "Query";
    movies: Array<{
        __typename?: "Movie";
        _id: number;
        title: string;
        vote_average: number;
        release_date: Date;
        runtime: number;
        poster_path?: string | null;
    }>;
};

export type GetMovieQueryVariables = Exact<{
    movieId: Scalars["Int"]["input"];
}>;

export type GetMovieQuery = {
    __typename?: "Query";
    movie?: {
        __typename?: "Movie";
        _id: number;
        title: string;
        vote_average: number;
        vote_count: number;
        status: string;
        release_date: Date;
        revenue: number;
        runtime: number;
        backdrop_path?: string | null;
        budget: number;
        homepage?: string | null;
        imdb_id?: string | null;
        original_language: string;
        original_title: string;
        overview: string;
        popularity: number;
        poster_path?: string | null;
        tagline?: string | null;
        genres: Array<string>;
        production_companies: Array<string>;
        production_countries: Array<string>;
        spoken_languages: Array<string>;
        keywords: Array<string>;
        reviews: Array<{
            __typename?: "Review";
            _id: string;
            username: string;
            rating: number;
            comment: string;
            date: Date;
        }>;
    } | null;
};

export type GetFiltersQueryVariables = Exact<{ [key: string]: never }>;

export type GetFiltersQuery = {
    __typename?: "Query";
    filters: {
        __typename?: "Filters";
        Genre: Array<string>;
        Rating: Array<string>;
        Decade: Array<string>;
        Status: Array<string>;
        Runtime: Array<string>;
    };
};

export type AddReviewMutationVariables = Exact<{
    movieId: Scalars["Int"]["input"];
    username: Scalars["String"]["input"];
    rating: Scalars["Int"]["input"];
    comment: Scalars["String"]["input"];
}>;

export type AddReviewMutation = {
    __typename?: "Mutation";
    addReview: {
        __typename?: "Movie";
        reviews: Array<{
            __typename?: "Review";
            _id: string;
            username: string;
            rating: number;
            comment: string;
            date: Date;
        }>;
    };
};
