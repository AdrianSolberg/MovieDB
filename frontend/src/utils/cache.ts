import { makeVar } from "@apollo/client";
import { Filters, SortingType } from "@/types/movieTypes";

export const sortOptionVar = makeVar<SortingType>(
    (sessionStorage.getItem("sort_option") as SortingType) ??
        SortingType.NEWEST_FIRST
);


const storedFilters = sessionStorage.getItem("filters");
const initialFilters: Filters = storedFilters ? JSON.parse(storedFilters) : {
    Genre: [],
    Rating: [],
    Decade: [],
    Status: [],
    Runtime: [],
};
export const filtersVar = makeVar<Filters>(initialFilters);

export const searchVar = makeVar<string>(
    sessionStorage.getItem("search") ?? ""
);

export const usernameVar = makeVar<string>(
    localStorage.getItem("username") ?? "Guest"
);
