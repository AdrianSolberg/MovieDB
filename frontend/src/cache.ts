import { makeVar } from "@apollo/client";
import { SortingType } from "./utils/searchSortAndFilter";

export const sortOptionVar = makeVar<SortingType>(
    (sessionStorage.getItem("sort_option") as SortingType) ??
        SortingType.NEWEST_FIRST
);

export const filtersVar = makeVar<{ [key: string]: string[] }>(
    JSON.parse(sessionStorage.getItem("filters") ?? "{}")
);

export const searchVar = makeVar<string>(
    sessionStorage.getItem("search") ?? ""
);
