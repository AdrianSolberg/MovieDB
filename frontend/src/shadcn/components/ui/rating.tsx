"use client";

import React from "react";
import { Star } from "lucide-react";

import { cn } from "../../lib/utils";

const ratingVariants = {
    default: {
        star: "text-foreground",
        emptyStar: "text-muted-foreground",
    },
    destructive: {
        star: "text-destructive",
        emptyStar: "text-destructive/70",
    },
    yellow: {
        star: "text-yellow-500",
        emptyStar: "text-yellow-200",
    },
};

interface RatingsProps extends React.HTMLAttributes<HTMLDivElement> {
    totalstars?: number;
    size?: number;
    fill?: boolean;
    Icon?: React.ReactElement;
    variant?: keyof typeof ratingVariants;
    asInput?: boolean;
    value: number;
    onValueChange?: (value: number) => void;
}

const Ratings = ({ ...props }: RatingsProps) => {
    const {
        totalstars = 5,
        size = 20,
        fill = true,
        Icon = <Star />,
        variant = "default",
        asInput = false,
        onValueChange,
        value,
        ...extraProps
    } = props;

    const handleKeyDown = (e: React.KeyboardEvent, index: number) => {
        if (asInput && e.key === " ") {
            // " " is the encoding for spacebar
            e.preventDefault();
            onValueChange && onValueChange(index + 1);
        }
    };

    const ratings = value;

    const fullStars = Math.floor(ratings);
    const partialStar =
        ratings % 1 > 0 ? (
            <PartialStar
                fillPercentage={ratings % 1}
                size={size}
                className={cn(ratingVariants[variant].star)}
                Icon={Icon}
                asInput={asInput}
                onValueChange={() =>
                    onValueChange && onValueChange(fullStars + 1)
                }
            />
        ) : null;

    return (
        <div
            className={cn("flex items-center gap-2")}
            role={asInput ? "radiogroup" : undefined}
            {...extraProps}
        >
            {[...Array(fullStars)].map((_, i) =>
                React.cloneElement(Icon, {
                    key: i,
                    size,
                    className: cn(
                        fill ? "fill-current" : "fill-transparent",
                        ratingVariants[variant].star,
                        asInput ? "cursor-pointer" : ""
                    ),
                    role: asInput ? "radio" : undefined,
                    "aria-checked": asInput ? i + 1 === value : undefined,
                    "aria-label": asInput ? `${i + 1} star` : undefined,
                    tabIndex: asInput ? 0 : -1,
                    onClick: () => onValueChange && onValueChange(i + 1),
                    onKeyDown: asInput
                        ? (e: React.KeyboardEvent) => handleKeyDown(e, i)
                        : undefined,
                })
            )}
            {partialStar}
            {[...Array(totalstars - fullStars - (partialStar ? 1 : 0))].map(
                (_, i) =>
                    React.cloneElement(Icon, {
                        key: i + fullStars + 1,
                        size,
                        className: cn(
                            ratingVariants[variant].emptyStar,
                            asInput ? "cursor-pointer" : ""
                        ),
                        role: asInput ? "radio" : undefined,
                        "aria-checked": asInput
                            ? fullStars + i + 1 + (partialStar ? 1 : 0) ===
                              value
                            : undefined,
                        tabIndex: asInput ? 0 : -1,
                        onKeyDown: asInput
                            ? (e: React.KeyboardEvent) =>
                                  handleKeyDown(
                                      e,
                                      fullStars + i + (partialStar ? 1 : 0)
                                  )
                            : undefined,
                        "aria-label": asInput
                            ? `${fullStars + i + 1 + (partialStar ? 1 : 0)} star`
                            : undefined,
                        onClick: () =>
                            onValueChange &&
                            onValueChange(
                                fullStars + i + 1 + (partialStar ? 1 : 0)
                            ),
                    })
            )}
        </div>
    );
};

interface PartialStarProps {
    fillPercentage: number;
    size: number;
    className?: string;
    Icon: React.ReactElement;
    asInput?: boolean;
    onValueChange?: () => void;
}

const PartialStar = ({ ...props }: PartialStarProps) => {
    const { fillPercentage, size, className, Icon, asInput, onValueChange } =
        props;

    return (
        <div
            role={asInput ? "input" : undefined}
            onClick={() => onValueChange && onValueChange()}
            className={cn("relative inline-block", asInput && "cursor-pointer")}
        >
            {React.cloneElement(Icon, {
                size,
                className: cn("fill-transparent", className),
            })}
            <div
                style={{
                    position: "absolute",
                    top: 0,
                    overflow: "hidden",
                    width: `${fillPercentage * 100}%`,
                }}
            >
                {React.cloneElement(Icon, {
                    size,
                    className: cn("fill-current", className),
                })}
            </div>
        </div>
    );
};

export default Ratings;
