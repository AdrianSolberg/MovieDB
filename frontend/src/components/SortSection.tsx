import {
    RadioGroup,
    RadioGroupItem,
} from "../shadcn/components/ui/radio-group";
import {
    AccordionContent,
    AccordionItem,
    AccordionTrigger,
} from "../shadcn/components/ui/accordion";
import { Label } from "../shadcn/components/ui/label";
import { sortOptionVar } from "@/utils/cache";
import { useReactiveVar } from "@apollo/client";
import { getSortOptionDisplayName } from "@/utils/sortOptionUtil";
import { SortingType } from "@/types/__generated__/types";

interface SortSectionInterface {
    loading: boolean;
}

const SortSection: React.FC<SortSectionInterface> = ({ loading }) => {
    const sortOption = useReactiveVar(sortOptionVar);

    const updateSortOption = (option: SortingType) => {
        sessionStorage.setItem("sort_option", option);
        sortOptionVar(option);
    };

    return (
        <AccordionItem value={`Sort item`}>
            <AccordionTrigger>
                <p>
                    Sort by{" "}
                    <span className="opacity-60">
                        ({getSortOptionDisplayName(sortOption)})
                    </span>
                </p>
            </AccordionTrigger>
            <AccordionContent>
                <RadioGroup value={sortOption} onValueChange={updateSortOption}>
                    <ul className="grid gap-2">
                        {Object.values(SortingType).map((option) => (
                            <li
                                key={option}
                                className="flex items-center space-x-2"
                            >
                                <RadioGroupItem
                                    value={option}
                                    id={option}
                                    disabled={loading}
                                    aria-label={option}
                                />
                                <Label
                                    htmlFor={option}
                                    className="hover:cursor-pointer"
                                >
                                    {getSortOptionDisplayName(option)}
                                </Label>
                            </li>
                        ))}
                    </ul>
                </RadioGroup>
            </AccordionContent>
        </AccordionItem>
    );
};

export default SortSection;
