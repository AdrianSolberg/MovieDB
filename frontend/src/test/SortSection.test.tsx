import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { vi } from "vitest";
import { useReactiveVar } from "@apollo/client";
import { sortOptionVar } from "@/utils/cache";
import { SortingType } from "@/types/__generated__/types";
import SortSection from "@/components/SortSection";
import { Accordion } from "@/shadcn/components/ui/accordion";
import "@testing-library/jest-dom";

vi.mock("@apollo/client", () => ({
    ...vi.importActual("@apollo/client"),
    useReactiveVar: vi.fn(),
}));

vi.mock("@/utils/cache", () => ({
    sortOptionVar: vi.fn(),
}));

vi.mock("@/utils/sortOptionUtil", () => ({
    getSortOptionDisplayName: vi.fn((option) => `Display name ${option}`),
}));

describe("SortSection Component", () => {
    const mockSortOption = SortingType.MOST_POPULAR;
    const mockUpdateSortOption = vi.fn();

    const renderSortSection = (loading = false) => {
        return render(
            <Accordion type="single" collapsible className="w-full">
                <SortSection loading={loading} />
            </Accordion>
        );
    };

    beforeEach(() => {
        vi.mocked(useReactiveVar).mockReturnValue(mockSortOption);
        vi.mocked(sortOptionVar).mockImplementation(mockUpdateSortOption);
        sessionStorage.clear();
        vi.clearAllMocks();
    });

    it("renders with correct default sort option", () => {
        renderSortSection();

        expect(screen.getByText("Sort by")).toBeInTheDocument();
        expect(
            screen.getByText(`(Display name ${mockSortOption})`)
        ).toBeInTheDocument();
    });

    it("displays all sorting options when expanded", async () => {
        renderSortSection();
        await userEvent.click(screen.getByRole("button"));

        const radioButtons = screen.getAllByRole("radio");
        expect(radioButtons).toHaveLength(Object.values(SortingType).length);

        Object.values(SortingType).forEach((option) => {
            const radioLabel = screen.getByText(`Display name ${option}`);
            expect(radioLabel).toBeInTheDocument();
            expect(radioLabel).not.toBeDisabled();
        });
    });

    it("updates sort option on selection", async () => {
        renderSortSection();
        await userEvent.click(screen.getByRole("button"));

        const newOption = SortingType.NEWEST_FIRST;
        const radioLabel = screen.getByText(`Display name ${newOption}`);
        await userEvent.click(radioLabel);

        expect(mockUpdateSortOption).toHaveBeenCalledWith(newOption);
        expect(sessionStorage.getItem("sort_option")).toBe(newOption);
    });

    it("disables radio buttons when loading", async () => {
        renderSortSection(true);
        await userEvent.click(screen.getByRole("button"));

        const radioButtons = screen.getAllByRole("radio");
        radioButtons.forEach((radio) => {
            expect(radio).toBeDisabled();
        });
    });

    it("reflects initial sort option from sessionStorage", async () => {
        const initialSortOption = SortingType.BEST_RATED;
        vi.mocked(useReactiveVar).mockReturnValue(initialSortOption);
        sessionStorage.setItem("sort_option", initialSortOption);

        renderSortSection();
        await userEvent.click(screen.getByRole("button"));

        const radioButtons = screen.getAllByRole("radio");
        radioButtons.forEach((radio) => {
            if (radio.id === initialSortOption) {
                expect(radio).toBeChecked();
                return;
            }
            expect(radio).not.toBeChecked();
        });

        expect(
            screen.getByText(`(Display name ${initialSortOption})`)
        ).toBeInTheDocument();
    });
});
