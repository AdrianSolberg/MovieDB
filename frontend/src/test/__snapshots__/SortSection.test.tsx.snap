// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SortSection > matches snapshot when collapsed 1`] = `
<DocumentFragment>
  <div
    class="w-full"
    data-orientation="vertical"
  >
    <div
      class="border-b"
      data-orientation="vertical"
      data-state="closed"
    >
      <h3
        class="flex"
        data-orientation="vertical"
        data-state="closed"
      >
        <button
          aria-controls="radix-:r1:"
          aria-expanded="false"
          class="flex flex-1 items-center justify-between py-4 font-medium transition-all hover:underline [&[data-state=open]>svg]:rotate-180"
          data-orientation="vertical"
          data-radix-collection-item=""
          data-state="closed"
          id="radix-:r0:"
          type="button"
        >
          <p>
            Sort by 
            <span
              class="opacity-60"
            >
              (Display name MOST_POPULAR)
            </span>
          </p>
          <svg
            class="lucide lucide-chevron-down h-4 w-4 shrink-0 transition-transform duration-200"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m6 9 6 6 6-6"
            />
          </svg>
        </button>
      </h3>
      <div
        aria-labelledby="radix-:r0:"
        class="max-h-40 overflow-y-auto text-sm transition-all data-[state=closed]:animate-accordion-up data-[state=open]:animate-accordion-down"
        data-orientation="vertical"
        data-state="closed"
        hidden=""
        id="radix-:r1:"
        role="region"
        style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`SortSection > matches snapshot when open 1`] = `
<DocumentFragment>
  <div
    class="w-full"
    data-orientation="vertical"
  >
    <div
      class="border-b"
      data-orientation="vertical"
      data-state="open"
    >
      <h3
        class="flex"
        data-orientation="vertical"
        data-state="open"
      >
        <button
          aria-controls="radix-:r3:"
          aria-expanded="true"
          class="flex flex-1 items-center justify-between py-4 font-medium transition-all hover:underline [&[data-state=open]>svg]:rotate-180"
          data-orientation="vertical"
          data-radix-collection-item=""
          data-state="open"
          id="radix-:r2:"
          type="button"
        >
          <p>
            Sort by 
            <span
              class="opacity-60"
            >
              (Display name MOST_POPULAR)
            </span>
          </p>
          <svg
            class="lucide lucide-chevron-down h-4 w-4 shrink-0 transition-transform duration-200"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m6 9 6 6 6-6"
            />
          </svg>
        </button>
      </h3>
      <div
        aria-labelledby="radix-:r2:"
        class="max-h-40 overflow-y-auto text-sm transition-all data-[state=closed]:animate-accordion-up data-[state=open]:animate-accordion-down"
        data-orientation="vertical"
        data-state="open"
        id="radix-:r3:"
        role="region"
        style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
      >
        <div
          class="pb-4 pt-0"
        >
          <div
            aria-required="false"
            class="grid gap-2"
            dir="ltr"
            role="radiogroup"
            style="outline: none;"
            tabindex="0"
          >
            <ul
              class="grid gap-2"
            >
              <li
                class="flex items-center space-x-2"
              >
                <button
                  aria-checked="true"
                  aria-label="MOST_POPULAR"
                  class="aspect-square h-4 w-4 rounded-full border border-primary text-primary ring-offset-background focus:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
                  data-radix-collection-item=""
                  data-state="checked"
                  id="MOST_POPULAR"
                  role="radio"
                  tabindex="-1"
                  type="button"
                  value="MOST_POPULAR"
                >
                  <span
                    class="flex items-center justify-center"
                    data-state="checked"
                  >
                    <svg
                      class="lucide lucide-circle h-2.5 w-2.5 fill-current text-current"
                      fill="none"
                      height="24"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <circle
                        cx="12"
                        cy="12"
                        r="10"
                      />
                    </svg>
                  </span>
                </button>
                <label
                  class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70 hover:cursor-pointer"
                  for="MOST_POPULAR"
                >
                  Display name MOST_POPULAR
                </label>
              </li>
              <li
                class="flex items-center space-x-2"
              >
                <button
                  aria-checked="false"
                  aria-label="NEWEST_FIRST"
                  class="aspect-square h-4 w-4 rounded-full border border-primary text-primary ring-offset-background focus:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
                  data-radix-collection-item=""
                  data-state="unchecked"
                  id="NEWEST_FIRST"
                  role="radio"
                  tabindex="-1"
                  type="button"
                  value="NEWEST_FIRST"
                />
                <label
                  class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70 hover:cursor-pointer"
                  for="NEWEST_FIRST"
                >
                  Display name NEWEST_FIRST
                </label>
              </li>
              <li
                class="flex items-center space-x-2"
              >
                <button
                  aria-checked="false"
                  aria-label="OLDEST_FIRST"
                  class="aspect-square h-4 w-4 rounded-full border border-primary text-primary ring-offset-background focus:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
                  data-radix-collection-item=""
                  data-state="unchecked"
                  id="OLDEST_FIRST"
                  role="radio"
                  tabindex="-1"
                  type="button"
                  value="OLDEST_FIRST"
                />
                <label
                  class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70 hover:cursor-pointer"
                  for="OLDEST_FIRST"
                >
                  Display name OLDEST_FIRST
                </label>
              </li>
              <li
                class="flex items-center space-x-2"
              >
                <button
                  aria-checked="false"
                  aria-label="BEST_RATED"
                  class="aspect-square h-4 w-4 rounded-full border border-primary text-primary ring-offset-background focus:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
                  data-radix-collection-item=""
                  data-state="unchecked"
                  id="BEST_RATED"
                  role="radio"
                  tabindex="-1"
                  type="button"
                  value="BEST_RATED"
                />
                <label
                  class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70 hover:cursor-pointer"
                  for="BEST_RATED"
                >
                  Display name BEST_RATED
                </label>
              </li>
              <li
                class="flex items-center space-x-2"
              >
                <button
                  aria-checked="false"
                  aria-label="WORST_RATED"
                  class="aspect-square h-4 w-4 rounded-full border border-primary text-primary ring-offset-background focus:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
                  data-radix-collection-item=""
                  data-state="unchecked"
                  id="WORST_RATED"
                  role="radio"
                  tabindex="-1"
                  type="button"
                  value="WORST_RATED"
                />
                <label
                  class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70 hover:cursor-pointer"
                  for="WORST_RATED"
                >
                  Display name WORST_RATED
                </label>
              </li>
              <li
                class="flex items-center space-x-2"
              >
                <button
                  aria-checked="false"
                  aria-label="LONGEST_RUNTIME"
                  class="aspect-square h-4 w-4 rounded-full border border-primary text-primary ring-offset-background focus:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
                  data-radix-collection-item=""
                  data-state="unchecked"
                  id="LONGEST_RUNTIME"
                  role="radio"
                  tabindex="-1"
                  type="button"
                  value="LONGEST_RUNTIME"
                />
                <label
                  class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70 hover:cursor-pointer"
                  for="LONGEST_RUNTIME"
                >
                  Display name LONGEST_RUNTIME
                </label>
              </li>
              <li
                class="flex items-center space-x-2"
              >
                <button
                  aria-checked="false"
                  aria-label="SHORTEST_RUNTIME"
                  class="aspect-square h-4 w-4 rounded-full border border-primary text-primary ring-offset-background focus:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
                  data-radix-collection-item=""
                  data-state="unchecked"
                  id="SHORTEST_RUNTIME"
                  role="radio"
                  tabindex="-1"
                  type="button"
                  value="SHORTEST_RUNTIME"
                />
                <label
                  class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70 hover:cursor-pointer"
                  for="SHORTEST_RUNTIME"
                >
                  Display name SHORTEST_RUNTIME
                </label>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
